extensions/amp-list/0.1/amp-list.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {UrlReplacements} from '../../../src/url-replacements';
+import {assertHttpsUrl} from '../../../src/url';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {templatesFor} from '../../../src/template';
+import {xhrFor} from '../../../src/xhr';
+
+/** @const {!Function} */
+const assert = AMP.assert;
+
+
+/**
+ */
+export class AmpList extends AMP.BaseElement {
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    /** @const {!Element} */
+    this.container_ = this.getWin().document.createElement('div');
+    this.applyFillContent(this.container_, true);
+    this.element.appendChild(this.container_);
+
+    /** @private @const {!UrlReplacements} */
+    this.urlReplacements_ = new UrlReplacements(this.getWin());
+  }
+
+  /** @override */
+  layoutCallback() {
+    const src = this.urlReplacements_.expand(assertHttpsUrl(
+        this.element.getAttribute('data-src'), this.element));
+    const opts = {
+      credentials: this.element.getAttribute('data-credentials')
+    };
+    return xhrFor(this.getWin()).fetchJson(src, opts).then(data => {
+      assert(typeof data == 'object' && Array.isArray(data['items']),
+          'Response must be {items: []} object %s %s',
+          this.element, data);
+      const items = data['items'];
+      return templatesFor(this.getWin()).findAndRenderTemplateArray(
+          this.element, items).then(this.rendered_.bind(this));
+    });
+  }
+
+  /**
+   * @param {!Array<!Element>} elements
+   * @private
+   */
+  rendered_(elements) {
+    elements.forEach(element => {
+      this.container_.appendChild(element);
+    });
+
+    // Change height if needed.
+    this.getVsync().measure(() => {
+      const scrollHeight = this.container_./*OK*/scrollHeight;
+      const height = this.element./*OK*/offsetHeight;
+      if (scrollHeight > height) {
+        this.requestChangeHeight(scrollHeight, newHeight => {
+          // TODO(dvoytenko): Implement fallback.
+        });
+      }
+    });
+  }
+}
+
+AMP.registerElement('amp-list', AmpList);
```

Comment:
You mean, do we need an `amp-list` vs a generic `amp-render`? If so, here are the reasons why I thought this was useful:

1. It's actually more direct with the original request
2. Clear how to extend it to an infinite scroll case (someone mentioned that as well)
3. Semantics of changing height are more clear
4. I expect as follow up we can add similar strategy to `amp-carousel` and such